src/lib/y2storage/blk_device.rb

```diff
@@ -49,13 +52,21 @@ def encrypted?
       !encryption.nil?
     end
 
-    # Filesystem placed in the device, either directly or through an encryption
-    # layer.
+    # Filesystem in the device
+    #
+    # By default it returns a filesystem placed either directly on the device
+    # or through an encryption layer.
+    #
+    # It can be forced to return only the direct filesystem (no encryption or any
+    # other layer in between).
     #
-    # @return [Filesystems::BlkFilesystem] nil if neither the raw device or its
-    #   encrypted version are formatted
-    def plain_blk_filesystem
-      encrypted? ? encryption.blk_filesystem : blk_filesystem
+    # @param [traverse_encryption] if set to false, the method will return nil
+    #   if the device is encrypted, no matter if the encrypted device is
+    #   formatted or not.
+    # @return [Filesystems::BlkFilesystem] nil if the device is not formatted
+    def blk_filesystem(traverse_encryption: true)
```

Comment:
what in fact I worry, that someone really use this argument for low level access `disk.blk_filesystem(traverse_encryption: false)` and then we introduce another `disk.blk_filesystem(traverse_encryption: false, traverse_cache: false)` for low level access and we need to change it everywhere. With my proposal there are just two cases

1. common user, want smart function

2. expert hacker want low-level